workers/unity/Assets/Gdk/Core/GameObjectRepresentation/EntityGameObjectLinker.cs

```diff
@@ -0,0 +1,29 @@
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class EntityGameObjectLinker
+    {
+        private readonly World world;
+
+        public EntityGameObjectLinker(World world)
+        {
+            this.world = world;
+        }
+
+        public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, long spatialEntityId,
+            ViewCommandBuffer viewCommandBuffer)
+        {
+            foreach (var component in gameObject.GetComponents<Component>())
+            {
+                viewCommandBuffer.AddComponent(entity, component.GetType(), component);
```

Comment:
If a GameObject has more than one of the same component, this will throw an exception I think (when it flushes)

Reply:
Yes, that's correct. Do we have a consensus on @samiwh's design doc? Happy to remove the behaviour altogether.

Reply:
If we aren't allowing multiple components then we should let the users know rather than letting Unity throw an arcane exception deep in the ECS

Reply:
Added a safeguard and a unit test case.